### PR TITLE
Add test for corner-case of having data_base64 as an extension in V03

### PR DESCRIPTION
**Pull request by markusthoemmes**
Adds a test that catches the corner-case of having a `data_base64` as an extension in V03.

Signed-off-by: Markus Thömmes <markusthoemmes@me.com>

### CODE REVIEW

File: v2/event/event_unmarshal_test.go [hunk 1]
```diff
@@ -829,6 +829,34 @@ func TestUnmarshalWithOrdering(t *testing.T) {
 				DataBase64:  true,
 			},
 		},
+		"base64 json encoded data v0.3 with data/data_base64 -> datacontenttype -> specversion -> datacontentencoding": {
```
Comment (slinkydeveloper) [inline v2/event/event_unmarshal_test.go:832]: Can you do a similar one without a base64 data (e.g. like struct data)

Comment (markusthoemmes) [reply v2/event/event_unmarshal_test.go]: Done!
